### DISCUSSION TITLE
extraconfig issue after upgrade cloudstack

### DISCUSSION THREAD

**Discussion by shulemmosko** [apache/cloudstack#9911] [Q&A] [answered]
##### ISSUE TYPE

 * Bug Report
 
##### COMPONENT NAME

KVM Extraconfig

##### CLOUDSTACK VERSION
4.18.2.4

##### SUMMARY

I had a VM with a vTPM configured in the settings as follow
![image](https://github.com/user-attachments/assets/52eaa472-f4f0-40eb-8549-f2a36dd609ee)

it was working fine untill i upgraded cloudstack to 4.18.2.4 after i upgraded cloudstack i cant start the VM im getting the following error 
![image](https://github.com/user-attachments/assets/768880a8-3e4e-422e-a9bb-9edd7f78ecc0)

and the extraconfig is graied out i cannot remove it or edit it.
![image](https://github.com/user-attachments/assets/5f3c17a6-1155-4f72-975c-90bc42bfc4a8)

##### EXPECTED RESULTS
the extraconfig device to work properly after upgrading cloudstack

**Comment by weizhouapache** [comment]
Due to security issue, users can no longer add extraconfig via vm settings.
Please use updatevm with extraconfig instead. It is not supported on UI for now, you can use cmk

**Comment by shulemmosko** [comment]
Ok got it thank you!

**Comment by shulemmosko** [comment]
How can i remove the extraconfig if its grayed out 
I need to be able to power on this vm 

Thanks!

**Comment by GutoVeronezi** [comment]
> How can i remove the extraconfig if its grayed out 
> I need to be able to power on this vm 
> 
> Thanks!

Hello @shulemmosko

You can try to remove the extraconfig by calling the updateVirtualMachine API via CMK with all the details you have in the VM, expect the extraconfig. For the whole syntax of the command, you might want to take a look at the browser's network tab when you update anything in the VM.

**Comment by weizhouapache** [comment]
> > How can i remove the extraconfig if its grayed out
> > I need to be able to power on this vm
> > Thanks!
> 
> Hello @shulemmosko
> 
> You can try to remove the extraconfig by calling the updateVirtualMachine API via CMK with all the details you have in the VM, expect the extraconfig. For the whole syntax of the command, you might want to take a look at the browser's network tab when you update anything in the VM.

I think user cannot delete extraconfig in that way any more.
the only way for now is, removing the record with name "extraconfig-1" from "user_vm_details" table.

**Comment by weizhouapache** [comment, marked as answer]
please keep eyes on #7849 #9878
